recipes/repo.rb

```diff
@@ -68,6 +68,67 @@
   end
 end
 
+# Configure gpg-vault
+user 'gpg-vault' do
+  manage_home true
+  comment 'GPG vault user'
+end
+execute 'gpg-init' do
+  command 'gpg -K'
+  environment 'HOME' => '/home/gpg-vault'
+  not_if { File.directory? '/home/gpg-vault/.gnupg' }
+  user 'gpg-vault'
+  group 'gpg-vault'
+end
+cookbook_file '/home/gpg-vault/.gnupg/gpg-agent.conf' do
+  source 'repo/gpg-agent.conf'
+  owner 'gpg-vault'
+  group 'gpg-vault'
+  mode '0600'
+end
+directory '/var/run/gpg-vault' do
+  owner 'gpg-vault'
+  group 'gpg-vault'
+  mode '0750'
+end
+cookbook_file '/etc/systemd/system/gpg-vault-agent.service' do
+  source 'gpg-vault-agent.service'
+end
+systemd_unit 'gpg-vault-agent.service' do
+  triggers_reload true
+  action [:start, :enable]
+end
+gpg_key = data_bag_item('ros_buildfarm_repository_signing_keys', node.chef_environment)
+file '/home/gpg-vault/.gnupg/gpg_public_key.pub' do
+  content gpg_key['public_key']
+  owner 'gpg-vault'
+  group 'gpg-vault'
+  mode '0644'
+end
+execute 'gpg --import /home/gpg-vault/.gnupg/gpg_public_key.pub' do
+  environment 'HOME' => '/home/gpg-vault'
+  user 'gpg-vault'
+  group 'gpg-vault'
+  not_if "gpg --list-keys #{gpg_key['fingerprint']}"
+end
+file '/home/gpg-vault/.gnupg/gpg_private_key.sec' do
+  content gpg_key['private_key']
+  owner 'gpg-vault'
+  group 'gpg-vault'
+  mode '0600'
+end
+execute 'gpg --import /home/gpg-vault/.gnupg/gpg_private_key.sec' do
+  environment 'HOME' => '/home/gpg-vault'
+  user 'gpg-vault'
+  group 'gpg-vault'
+  not_if "gpg --list-secret-keys #{gpg_key['fingerprint']}"
+end
+group 'gpg-vault' do
+  append true
+  members [agent_username]
+  action [:manage]
+end
```

Comment:
Does this need to be further up in the file, before it is referenced?

Reply:
I think the question is can we reference the gpg-vault group in the above resources before we use this resource?

If I'm wrong then you can ignore this and I would need clarification on the question.

We can reference it above because the gpg-agent group is created when the gpg-vault user is created based on default login.defs per this extraction from the useradd(8) manual

```
-g, --gid GROUP
 The group name or number of the user's initial login group. The group name must exist. A group number must refer to an already existing group.

 If not specified, the behavior of useradd will depend on the USERGROUPS_ENAB variable in /etc/login.defs. If this variable is set to yes (or -U/--user-group is specified on the command line), a group will be created for the user, with the same name as her loginname. If the variable is set to no (or
 -N/--no-user-group is specified on the command line), useradd will set the primary group of the new user to the value specified by the GROUP variable in /etc/default/useradd, or 100 by default.
```

Reply:
Yes, you answered my question. I understand the purpose of this snip now, thanks.